From 9a6e360c0f91998ebd41fe26b8bfc98c9b98acd6 Mon Sep 17 00:00:00 2001
From: Joel Hansen <joel.hansen@mailfence.com>
Date: Fri, 30 Aug 2024 09:37:20 +1000
Subject: [PATCH 1/1] Add hostname dependency to avahi-daemon service

Alpine uses the OpenRC services from Gentoo, but the service for
avahi-daemon doesn't need "hostname". For Alpine this causes avahi to
start before the host name is set, thus a host name of "none" is
advertised.

This patch ensures the OpenRC service installed on Alpine depends on
hostname.
---
 initscript/gentoo/avahi-daemon.in | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/initscript/gentoo/avahi-daemon.in b/initscript/gentoo/avahi-daemon.in
index 6a14b5b..c06006d 100644
--- a/initscript/gentoo/avahi-daemon.in
+++ b/initscript/gentoo/avahi-daemon.in
@@ -7,7 +7,7 @@ extra_started_commands="reload"
 depend() {
 	before netmount nfsmount
 	use net
-	need dbus
+	need dbus hostname
 }
 
 start() {
-- 
2.46.0

